bundles/org.eclipse.emfcloud.modelserver.emf/src/org/eclipse/emfcloud/modelserver/emf/common/codecs/CodecsManager.java

```diff
@@ -44,41 +44,45 @@ public interface CodecsManager {
    /**
     * Encode an EObject to JsonNodes in all available formats.
     *
-    * @param eObject
+    * @param modelUri The ModelUri of the request, null if not available
+    * @param eObject  The EObject to encode
     * @return a map containing all format strings and their encoded JsonNodes
     * @throws EncodingException
     */
-   Map<String, JsonNode> encode(EObject eObject) throws EncodingException;
+   Map<String, JsonNode> encode(String modelUri, EObject eObject) throws EncodingException;
 
    /**
     * Encode an EObject to a JsonNode.
     *
-    * @param context the javalin http context
-    * @param eObject EObject to encode
+    * @param modelUri The ModelUri of the request, null if not available
+    * @param context  the javalin http context
+    * @param eObject  EObject to encode
     * @return JsonNode
     * @throws EncodingException when encoding failed
     */
-   JsonNode encode(Context context, EObject eObject) throws EncodingException;
+   JsonNode encode(String modelUri, Context context, EObject eObject) throws EncodingException;
 
    /**
     * Encode an EObject to a JsonNode.
     *
-    * @param context the javalin websocket context
-    * @param eObject EObject to encode
+    * @param modelUri The ModelUri of the request, null if not available
+    * @param context  the javalin websocket context
+    * @param eObject  EObject to encode
     * @return JsonNode
     * @throws EncodingException when encoding failed
     */
-   JsonNode encode(WsContext context, EObject eObject) throws EncodingException;
+   JsonNode encode(String modelUri, WsContext context, EObject eObject) throws EncodingException;
 
    /**
     * Decode a JsonNode to an EObject.
     *
-    * @param context the javalin http context
-    * @param payload tthe String payload holding EObject definition to decode
+    * @param modelUri The ModelUri of the request, null if not available
+    * @param context  the javalin http context
+    * @param payload  tthe String payload holding EObject definition to decode
```

Comment:
```suggestion
    * @param payload  the String payload holding EObject definition to decode
```
and the other occurrences in this file.